parse.go

```diff
@@ -33,18 +33,19 @@ func parse(templateName string, templateMap map[string]string, data any) ([]byte
 	return buf.Bytes(), nil
 }
 
-// withTemplates - load templates by file path
-func withTemplates(dirPath string, fileSuffix string) (map[string]string, error) {
+// withTemplates - load templates by directory entry
+func withTemplates(fsys fs.FS, dirPath, fileSuffix string) (map[string]string, error) {
 	rootTemplates := map[string]string{}
-	files, err := os.ReadDir(dirPath)
+
```

Comment:
What's the diff here, between `os` and `fs` ? 
(asking coz i don't know :P )

Reply:
All good, just read the docs :P

Reply:
`os` uses the actual directory, whereas with `fs` you can specify which file system you want to use
`os.ReadFile()` was causing issues when using the `embed.FS` and wasn't reading the embedded file